code/modules/mob/living/carbon/human/life/handle_disabilities.dm

```diff
@@ -1,6 +1,26 @@
 //Refer to life.dm for caller
 
 /mob/living/carbon/human/proc/handle_disabilities()
+	if(disabilities & ELECTROSENSE)
+		var/affect_chance = 100
+		if(head && istype(head,/obj/item/clothing/head/tinfoil))
+			affect_chance /= 2
+		if(wear_suit && istype(wear_suit,/obj/item/clothing/suit/spaceblanket))
+			affect_chance /= 2
+		for(var/obj/machinery/M in range(3,src))
+			if(!(M.stat & NOPOWER) && prob(affect_chance))
```

Comment:
I suggest 2 more checks for machinery.

The machine should also not be broken or disabled(`!(M.stat & (NOPOWER|BROKEN|FORCEDISABLE))`
The machine should draw power `use_power > 0` -- this is particularly important because pipes are non-power drawing machinery.

Additionally - but unrelated - I suggest you search for objects and mobs that have a cell. For example:

```
for(var/atom/A in range(3, src))
  if(!A.get_cell())
    continue
  //Bad effects here
```

This would mean that you could get burned by being near eguns, batons, mechs, cyborgs, mommis, hardsuits, etc. -- however, this is a SIGNIFICANT expansion of danger so this is merely a suggestion.

Reply:
did these, also only included cells if they had charge and limited the range a bit more for them